src/System.Text.Json/tests/WritableJsonApiTests.cs

```diff
@@ -624,6 +966,324 @@ public static void TestAquiringAllProperties()
+                    return true;
+                }
+               
+                if(right.GetType() != left.GetType())
+                {
+                    return false;
+                }
+
+                switch(left)
+                {
+                    case JsonObject leftJsonObject:
+                        var rightJsonObject = right as JsonObject;
+                        if (leftJsonObject.Count() != rightJsonObject.Count())
+                        {
+                            return false;
+                        }
+
+                        for(int idx = 0; idx < leftJsonObject.Count(); idx++)
+                        {
+                            KeyValuePair<string, JsonNode> leftElement = leftJsonObject.ElementAt(idx);
+                            KeyValuePair<string, JsonNode> rightElement = rightJsonObject.ElementAt(idx);
+
+                            if(leftElement.Key != rightElement.Key || !recursiveEquals(leftElement.Value, rightElement.Value))
+                            {
+                                return false;
+                            }
+                        }
+
+                        return true;
+                    case JsonArray leftJsonArray:
+                        var rightJsonArray = right as JsonArray;
+                        if (leftJsonArray.Count() != rightJsonArray.Count())
+                        {
+                            return false;
+                        }
+                        for (int idx = 0; idx < leftJsonArray.Count(); idx++)
+                        {
+                            JsonNode leftElement = leftJsonArray.ElementAt(idx);
+                            JsonNode rightElement = rightJsonArray.ElementAt(idx);
+
+                            if(!recursiveEquals(leftElement, rightElement))
+                            {
+                                return false;
+                            }
+                        }
+
+                        return true;
+                    case JsonString leftJsonString:
+                        return leftJsonString.Equals(right as JsonString);
+                    case JsonNumber leftJsonNumber:
+                        return leftJsonNumber.Equals(right as JsonNumber);
+                    case JsonBoolean leftJsonBoolean:
+                        return leftJsonBoolean.Equals(right as JsonBoolean);
+                    default:
+                        return false;
+                }
+            }
+
+            Assert.True(recursiveEquals(employee, employeeCopy));
+        }
+
+        /// <summary>
+        /// Checking IsImmutable property
+        /// </summary>
+        [Fact]
+        public static void TestIsImmutable()
+        {
+            JsonElement employee = Mailbox.RetrieveMutableEmployeeData();
+            Assert.False(employee.IsImmutable);
+
+            if (!employee.IsImmutable)
+            {
+                JsonObject employeeNode = JsonNode.GetNode(employee) as JsonObject;
```

Comment:
I'm not sure I understand what GetNode does.

Reply:
Here's a discussion on it: https://github.com/kasiabulat/corefx/pull/2#discussion_r305025863

Reply:
I suppose it is just a Find on the tree?

Reply:
JsonElement can have either JsonNode or JsonElement underneath. If it's JsonNode, this method returns it.